src/test/java/org/opentripplanner/openstreetmap/tagmapping/DefaultMapperTest.java

```diff
@@ -111,6 +117,32 @@ void stairs() {
     assertEquals(PEDESTRIAN, props.getPermission());
   }
 
+  @Test
+  void footDiscouraged() {
+    var regular = WayTestData.pedestrianTunnel();
+    var props = wps.getDataForWay(regular);
+    assertEquals(PEDESTRIAN_AND_BICYCLE, props.getPermission());
```

Comment:
It is a bit surprising that highway=footway gets BICYCLE permission by default, even when bicycle=no is set.  Anyway, that issue is out of the scope of this PR.

Reply:
Where do you see `bicycle=no`? https://github.com/opentripplanner/OpenTripPlanner/blob/3d17964b653d7fd3fde801a9a95f360888be014e/src/test/java/org/opentripplanner/openstreetmap/wayproperty/specifier/WayTestData.java#L24-L34

However, `highway=footway` should probably not allow bicycles: https://wiki.openstreetmap.org/wiki/Tag:highway=footway?uselang=en

We should indeed change it in a separate PR.

Reply:
I tested adding bicycle=no myself,  just for curiosity :) It did not change anything.

Reply:
I believe that it's this piece of code taking care of it: https://github.com/opentripplanner/OpenTripPlanner/blob/73d873b91c1bd0fd5f6511c15722cb2517b19e4b/src/main/java/org/opentripplanner/graph_builder/module/osm/OsmModule.java#L250-L256

I would be in favour of moving that somehow together with the other code figuring out the permissions.

Reply:
I started working on that, and also fixing some known errors in traversal permission processing.